### PR TITLE
added access 

### PR DESCRIPTION
**Pull request by Pelumi527**

### CODE REVIEW

File: smart-contract/contracts/ElectionFactory.sol [hunk 1]
```diff
@@ -74,5 +74,10 @@ contract ElectionFactory {
         electionCount = count;
 
         emit CreateElection(count, election, msg.sender, _position);
+        return address(election);
+    }
+
+    function allElections() public view returns(ElectionDetails[] memory){
+        return elections;
     }
```
Comment (prince-curie) [inline smart-contract/contracts/ElectionFactory.sol]: This is made to be paginated and worked on by flowzz. Also make the method external

File: smart-contract/contracts/ElectionFactory.sol [hunk 2]
```diff
@@ -54,11 +54,11 @@ contract ElectionFactory {
...
 
     /// @dev Deploys a new election smart contract and stores the details.
-    function createElection(string memory _position, string[] memory _contestants) external onlyOwner {
+    function createElection(string memory _position, string[] memory _contestants, address[] memory _contestantsAddress) external onlyOwner returns(address) {
```
Comment (prince-curie) [inline smart-contract/contracts/ElectionFactory.sol]: No need to return the address you can pick the address up from the event emitted

File: smart-contract/contracts/ElectionFactory.sol [hunk 3]
```diff
@@ -54,11 +54,11 @@ contract ElectionFactory {
...
         uint256 count = electionCount;
         count++;
 
-        Election election = new Election(msg.sender, _position, _contestants.length, _contestants, count);
+        Election election = new Election(msg.sender, _position, _contestants.length, _contestants, _contestantsAddress);
```
Comment (prince-curie) [inline smart-contract/contracts/ElectionFactory.sol]: Please as discussed yesterday remove the contestant address? This looks like we are trying to dox our contestants.

Comment (Pelumi527) [reply smart-contract/contracts/ElectionFactory.sol]: we need the address, we need something unique to recognize each contestant, Name can be the same but we can't have two address

Comment (prince-curie) [reply smart-contract/contracts/ElectionFactory.sol]: Collect their email please.

**Pull request closed without merging**